build.gradle

```diff
@@ -154,7 +154,10 @@ dependencies {
     implementation("org.apache.logging.log4j:log4j-jul:${log4jVersion}")
     implementation("org.opensearch.client:opensearch-rest-high-level-client:${opensearchVersion}")
     api("org.opensearch.client:opensearch-rest-client:${opensearchVersion}")
-    api("org.opensearch.client:opensearch-java:${opensearchVersion}")
+    api("org.opensearch.client:opensearch-java:${opensearchVersion}") {
```

Comment:
The Apache HttpClient5 are coming from `opensearch-rest-client` and  `opensearch-java`, to disambiguate any conflicts, I would suggest to rely on `opensearch-rest-client`

Reply:
Looks like `httpclient5:5.2.1` is bringing `httpcore5:5.2` instead of `5.2.2` causing the conflict in the first place.
```
org.apache.httpcomponents.core5:httpcore5:5.2 -> 5.2.2
\--- org.apache.httpcomponents.client5:httpclient5:5.2.1
     +--- org.opensearch.client:opensearch-rest-client:3.0.0-SNAPSHOT
     |    +--- runtimeClasspath
     |    \--- org.opensearch.client:opensearch-rest-high-level-client:3.0.0-SNAPSHOT
     |         \--- runtimeClasspath
     \--- org.opensearch.client:opensearch-java:3.0.0-SNAPSHOT
          \--- runtimeClasspath
```

Reply:
This is quite strange, I will double check that but the deps are consistent (in Gradle at least):

```
  api "org.apache.httpcomponents.client5:httpclient5:${versions.httpclient5}"
  api "org.apache.httpcomponents.core5:httpcore5:${versions.httpcore5}"
  api "org.apache.httpcomponents.core5:httpcore5-h2:${versions.httpcore5}"
```

Reply:
> Btw, why do we need 2 clients here?

That's a good question. When we started building SDK we initially had java client which was causing a bunch of issues for the migration of a plugin to an extension. Later, on we moved to HLRC.
There's an open issue https://github.com/opensearch-project/opensearch-sdk-java/issues/732 discussing about redesigning the SDKClient and just have HLRC here. Java Client is something extension can include in their own build.gradle.

Reply:
Just for the sake of it (you know that), `opensearch-java` is what we recommend and it also has no dependencies on any modules from OpenSearch core (in 3.x release line)

Reply:
Yes, that would be the right client to go about it. It's just all the plugins are written based on NodeClient, converting those parsers and others would be a task while migrating from a plugin to an extension.

Reply:
> Just for the sake of it (you know that), `opensearch-java` is what we recommend and it also has no dependencies on any modules from OpenSearch core (in 3.x release line)

Yep, long discussion in #294 as Owais mentioned, and also specific plans in #302.

The idea to remove the clients as an API dependency is associated with multithreading and resource leakage concerns.